File: leetcode/q110/n.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def depth(self, root: TreeNode) -> int:
        if root is None:
            return 0
        return max(1 + self.depth(root.left), 1+self.depth(root.right))

    def isBalanced(self, root: TreeNode) -> bool:
        if root is None:
            return True
        flag = True if abs(self.depth(root.left)-self.depth(root.right)) <= 1 else False
        return flag and self.isBalanced(root.left) and self.isBalanced(root.right)

File: leetcode/q235/n.cpp
#include<iostream>
#include<vector>
#include<cstdio>
#include<cstring>
#include<algorithm>
#define ll long long 

using namespace std;

 struct TreeNode {
     int val;
     TreeNode *left;
     TreeNode *right;
     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 };

class Solution {
public:
    TreeNode* lowestCommonAncestor(TreeNode* root, TreeNode* p, TreeNode* q) {
        int mx = max(p->val, q->val);
        int mn = min(p->val, q->val);
        if(root->val < mn) {
            return lowestCommonAncestor(root->right, p, q);
        }
        else if(root->val > mx)
            return lowestCommonAncestor(root->left, p, q);
        else
            return root;

    }
};
